lib/main/atom/autoCompleteProvider.ts

```diff
@@ -88,6 +89,53 @@ export class AutocompleteProvider implements Provider {
     return suggestions
   }
 
+  formatImportPath(sourcePath: string): string {
+    sourcePath = sourcePath.replace(/\.d$/, ""); // tsconfig.removeExt can convert d.ts file path into the filepath end with `.d`;
+    sourcePath = sourcePath.replace(/.*\/node_modules\//, "");
+    return sourcePath;
+  }
+
+  async getPathSuggestions(
+    prefix: string,
+    location: FileLocationQuery): Promise<SuggestionWithDetails[]> {
+    const client = await this.clientResolver.get(location.file)
+    const projectInfo = await client.executeProjectInfo({
+      file: location.file,
+      needFileNameList: true
+    })
+    const filePaths = projectInfo.body!.fileNames!.filter(p => p != location.file);
+    var files: {
+      name: string;
+      relativePath: string;
+      fullPath: string;
+    }[] = []
+    var sourceDir = path.dirname(location.file);
+    filePaths.forEach(p => {
+      let relativePath = path.relative(sourceDir, p).split('\\').join('/');
+      relativePath = relativePath.substr(0, relativePath.lastIndexOf("."));
+      if (relativePath[0] !== '.') {
+        relativePath = './' + relativePath;
+      }
+      files.push({
+        name: path.basename(p, '.ts'),
+        relativePath: this.formatImportPath(relativePath),
+        fullPath: p
+      });
+    })
+    files = fuzzaldrin.filter(files, prefix, {key: "name"})
+    return files.map(file => {
+      let suggestionText = file.relativePath;
+      let suggestion: Suggestion = {
+        text: suggestionText,
+        replacementPrefix: prefix,
+        rightLabelHTML: '<span>' + file.name + '</span>',
+        type: 'import'
+      }
+
+      return suggestion;
```

Comment:
This whole function is messy. You're mixing `const` and `var` and mapping over arrays multiple times and creating unnecessary intermediate objects. You should also separate the relative path logic from generating and filtering the suggestions.